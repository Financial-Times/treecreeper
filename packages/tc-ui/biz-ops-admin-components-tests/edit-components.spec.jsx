// const React = require('react');
// const schema = require('@financial-times/tc-schema-sdk');
// require('../../../src/lib/configure-schema');
// const {
// 	EditText,
// 	EditTextArea,
// 	EditDropdown,
// 	EditBoolean,
// 	EditDateTime,
// } = require('../../../src/templates/components/edit-components');

// const { render } = require('../../testHelpers/component');

// const sharedPropsWithValue = {
// 	propertyName: 'PROPERTY NAME',
// 	value: 'VALUE',
// 	template: 'EditText',
// 	dataType: 'DATA TYPE',
// 	options: [],
// 	type: 'SystemLifecycle',
// 	label: 'LABEL',
// 	description: 'DESCRIPTION',
// };

// const sharedPropsNoValue = {
// 	propertyName: 'PROPERTY NAME',
// 	value: '',
// 	template: 'EditText',
// 	dataType: 'DATA TYPE',
// 	options: [],
// 	type: 'SystemLifecycle',
// 	label: 'LABEL',
// 	description: 'DESCRIPTION',
// };

// const testComponentProperties = (testCase, Component, props) => {
// 	it(`Should display a populated ${testCase} element with values`, () => {
// 		const renderedComponent = render(<Component {...props} />);
// 		expect(renderedComponent).toMatchSnapshot();
// 	});

// 	it(`Should display a blank ${testCase} element with no values`, () => {
// 		props = sharedPropsNoValue;
// 		const renderedComponent = render(<Component {...props} />);
// 		expect(renderedComponent).toMatchSnapshot();
// 	});
// };
// describe('edit components', () => {
// 	beforeAll(() => schema.refresh());

// 	describe('<EditText />', () => {
// 		testComponentProperties('EditText', EditText, sharedPropsWithValue);
// 	});

// 	describe('<EditTextArea />', () => {
// 		testComponentProperties(
// 			'EditTextArea',
// 			EditTextArea,
// 			sharedPropsWithValue,
// 		);
// 	});

// 	describe('<EditDropdown />', () => {
// 		testComponentProperties(
// 			'EditDropdown',
// 			EditDropdown,
// 			sharedPropsWithValue,
// 		);

// 		it(`Should render an EditDropdown element with a set of options`, () => {
// 			const propsWithOptions = {
// 				propertyName: 'PROPERTY NAME',
// 				value: 'VALUE',
// 				template: 'EditText',
// 				dataType: 'DATA TYPE',
// 				options: ["Don't know", 'Option 1', 'Option 2'],
// 				type: 'SystemLifecycle',
// 				label: 'LABEL',
// 				description: 'DESCRIPTION',
// 			};
// 			const component = render(<EditDropdown {...propsWithOptions} />);
// 			expect(component).toMatchSnapshot();
// 		});
// 	});

// 	describe('<EditDateTime />', () => {
// 		it('returns date value (default)', () => {
// 			const sharedPropsWithDateValue = {
// 				propertyName: 'PROPERTY NAME',
// 				value: { formatted: '2019-01-25T15:47:20.015000000Z' },
// 				template: 'EditText',
// 				dataType: 'DATA TYPE',
// 				options: [],
// 				type: 'Date',
// 				label: 'LABEL',
// 				description: 'DESCRIPTION',
// 			};
// 			const component = render(
// 				<EditDateTime {...sharedPropsWithDateValue} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('returns time value', () => {
// 			const sharedPropsWithTimeValue = {
// 				propertyName: 'PROPERTY NAME',
// 				value: { formatted: '21:00' },
// 				template: 'EditText',
// 				dataType: 'DATA TYPE',
// 				options: [],
// 				type: 'Time',
// 				label: 'LABEL',
// 				description: 'DESCRIPTION',
// 			};
// 			const component = render(
// 				<EditDateTime {...sharedPropsWithTimeValue} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('returns date and time value', () => {
// 			const sharedPropsWithDateTimeValue = {
// 				propertyName: 'PROPERTY NAME',
// 				value: { formatted: '2019-01-25T15:47:20.015000000Z' },
// 				template: 'EditText',
// 				dataType: 'DATA TYPE',
// 				options: [],
// 				type: 'DateTime',
// 				label: 'LABEL',
// 				description: 'DESCRIPTION',
// 			};
// 			const component = render(
// 				<EditDateTime {...sharedPropsWithDateTimeValue} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});
// 	});

// 	describe('<EditBoolean />', () => {
// 		testComponentProperties(
// 			'EditBoolean',
// 			EditBoolean,
// 			sharedPropsWithValue,
// 		);
// 	});
// });
