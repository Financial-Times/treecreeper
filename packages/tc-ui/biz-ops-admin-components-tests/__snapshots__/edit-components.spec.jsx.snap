// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit components <EditBoolean /> Should display a blank EditBoolean element with no values 1`] = `
<div class="o-forms-field" data-biz-ops-type="boolean" role="group" aria-labelledby="inline-radio-round-group-title">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--radio-round o-forms-input--inline">
    <label>
      <input type="radio" name="PROPERTY NAME" value="true" aria-label="Yes"
      id="radio1-PROPERTY NAME" />
      <span class="o-forms-input__label" aria-hidden="true">Yes</span>
    </label>
    <label>
      <input type="radio" name="PROPERTY NAME" value="false" aria-label="No"
      id="radio2-PROPERTY NAME" />
      <span class="o-forms-input__label" aria-hidden="true">No</span>
    </label>
  </span>
</div>
`;

exports[`edit components <EditBoolean /> Should display a populated EditBoolean element with values 1`] = `
<div class="o-forms-field" data-biz-ops-type="boolean" role="group" aria-labelledby="inline-radio-round-group-title">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--radio-round o-forms-input--inline">
    <label>
      <input type="radio" name="PROPERTY NAME" value="true" aria-label="Yes"
      id="radio1-PROPERTY NAME" />
      <span class="o-forms-input__label" aria-hidden="true">Yes</span>
    </label>
    <label>
      <input type="radio" name="PROPERTY NAME" value="false" aria-label="No"
      id="radio2-PROPERTY NAME" />
      <span class="o-forms-input__label" aria-hidden="true">No</span>
    </label>
  </span>
</div>
`;

exports[`edit components <EditDateTime /> returns date and time value 1`] = `
<label class="o-forms-field" data-biz-ops-type="temporal">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--text">
    <input name="PROPERTY NAME" id="id-PROPERTY NAME" type="datetime-local"
    value="2019-01-25T15:47:20.015" />
  </span>
</label>
`;

exports[`edit components <EditDateTime /> returns date value (default) 1`] = `
<label class="o-forms-field" data-biz-ops-type="temporal">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--text">
    <input name="PROPERTY NAME" id="id-PROPERTY NAME" type="date" value="2019-01-25"
    />
  </span>
</label>
`;

exports[`edit components <EditDateTime /> returns time value 1`] = `
<label class="o-forms-field" data-biz-ops-type="temporal">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--text">
    <input name="PROPERTY NAME" id="id-PROPERTY NAME" type="time" value="21:00"
    />
  </span>
</label>
`;

exports[`edit components <EditDropdown /> Should display a blank EditDropdown element with no values 1`] = `
<label class="o-forms-field" data-biz-ops-type="enum">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
      <div data-o-component="o-expander" class="o-expander"
      data-o-expander-shrink-to="hidden" data-o-expander-collapsed-toggle-text="more info"
      data-o-expander-expanded-toggle-text="less">
        <button class="o-expander__toggle o--if-js" type="button">more info</button>
        <div class="o-expander__content">
          <p>The lifecycle stage of a system. Note that 'Deprecated' should be used
            for systems that are currently not actively used, but may not be fully
            decommissioned yet</p>
          <dl>
            <dt>Preproduction</dt>
            <dd>Not yet ready for use. May be unstable, subject to unannounced changes,
              or not yet fully built.</dd>
            <dt>Production</dt>
            <dd>Ready for use by its intended users, and supported appropriately.</dd>
            <dt>Deprecated</dt>
            <dd>In the process of being replaced or retired. May still be functional,
              but should not be added as a new dependency to systems.</dd>
            <dt>Discontinued</dt>
            <dd>No longer functional. Attempts to interact with the system will be unsuccessful.</dd>
            <dt>Decommissioned</dt>
            <dd>For a self hosted/installed system, all its parts have been removed from
              all the environments it ran in. For SaaS, we no longer pay for or have
              the ability to use the system.</dd>
          </dl>
        </div>
      </div>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--select">
    <select id="id-PROPERTY NAME" name="PROPERTY NAME">
      <option value="null" selected>Don&#x27;t know</option>
    </select>
  </span>
</label>
`;

exports[`edit components <EditDropdown /> Should display a populated EditDropdown element with values 1`] = `
<label class="o-forms-field" data-biz-ops-type="enum">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
      <div data-o-component="o-expander" class="o-expander"
      data-o-expander-shrink-to="hidden" data-o-expander-collapsed-toggle-text="more info"
      data-o-expander-expanded-toggle-text="less">
        <button class="o-expander__toggle o--if-js" type="button">more info</button>
        <div class="o-expander__content">
          <p>The lifecycle stage of a system. Note that 'Deprecated' should be used
            for systems that are currently not actively used, but may not be fully
            decommissioned yet</p>
          <dl>
            <dt>Preproduction</dt>
            <dd>Not yet ready for use. May be unstable, subject to unannounced changes,
              or not yet fully built.</dd>
            <dt>Production</dt>
            <dd>Ready for use by its intended users, and supported appropriately.</dd>
            <dt>Deprecated</dt>
            <dd>In the process of being replaced or retired. May still be functional,
              but should not be added as a new dependency to systems.</dd>
            <dt>Discontinued</dt>
            <dd>No longer functional. Attempts to interact with the system will be unsuccessful.</dd>
            <dt>Decommissioned</dt>
            <dd>For a self hosted/installed system, all its parts have been removed from
              all the environments it ran in. For SaaS, we no longer pay for or have
              the ability to use the system.</dd>
          </dl>
        </div>
      </div>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--select">
    <select id="id-PROPERTY NAME" name="PROPERTY NAME">
      <option value="null">Don&#x27;t know</option>
    </select>
  </span>
</label>
`;

exports[`edit components <EditDropdown /> Should render an EditDropdown element with a set of options 1`] = `
<label class="o-forms-field" data-biz-ops-type="enum">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
      <div data-o-component="o-expander" class="o-expander"
      data-o-expander-shrink-to="hidden" data-o-expander-collapsed-toggle-text="more info"
      data-o-expander-expanded-toggle-text="less">
        <button class="o-expander__toggle o--if-js" type="button">more info</button>
        <div class="o-expander__content">
          <p>The lifecycle stage of a system. Note that 'Deprecated' should be used
            for systems that are currently not actively used, but may not be fully
            decommissioned yet</p>
          <dl>
            <dt>Preproduction</dt>
            <dd>Not yet ready for use. May be unstable, subject to unannounced changes,
              or not yet fully built.</dd>
            <dt>Production</dt>
            <dd>Ready for use by its intended users, and supported appropriately.</dd>
            <dt>Deprecated</dt>
            <dd>In the process of being replaced or retired. May still be functional,
              but should not be added as a new dependency to systems.</dd>
            <dt>Discontinued</dt>
            <dd>No longer functional. Attempts to interact with the system will be unsuccessful.</dd>
            <dt>Decommissioned</dt>
            <dd>For a self hosted/installed system, all its parts have been removed from
              all the environments it ran in. For SaaS, we no longer pay for or have
              the ability to use the system.</dd>
          </dl>
        </div>
      </div>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--select">
    <select id="id-PROPERTY NAME" name="PROPERTY NAME">
      <option value="null">Don&#x27;t know</option>
      <option value="null">Don&#x27;t know</option>
      <option value="Option 1">Option 1</option>
      <option value="Option 2">Option 2</option>
    </select>
  </span>
</label>
`;

exports[`edit components <EditText /> Should display a blank EditText element with no values 1`] = `
<label class="o-forms-field" data-biz-ops-type="text">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--text">
    <input name="PROPERTY NAME" id="id-PROPERTY NAME" class="o-forms__text"
    type="text" />
  </span>
</label>
`;

exports[`edit components <EditText /> Should display a populated EditText element with values 1`] = `
<label class="o-forms-field" data-biz-ops-type="text">
  <span class="o-forms-title">
    <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
    <span class="o-forms-title__prompt description-text">
      <span>DESCRIPTION</span>
    </span>
  </span>
  <span class="o-forms-input o-forms-input--text">
    <input name="PROPERTY NAME" id="id-PROPERTY NAME" class="o-forms__text"
    type="text" value="VALUE" />
  </span>
</label>
`;

exports[`edit components <EditTextArea /> Should display a blank EditTextArea element with no values 1`] = `
<div class="">
  <label class="o-forms-field" data-biz-ops-type="textarea">
    <span class="o-forms-title">
      <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
      <span class="o-forms-title__prompt description-text">
        <span>DESCRIPTION</span>
      </span>
    </span>
    <span class="o-forms-input o-forms-input--textarea">
      <textarea name="PROPERTY NAME" id="id-PROPERTY NAME" rows="8"></textarea>
    </span>
  </label>
</div>
`;

exports[`edit components <EditTextArea /> Should display a populated EditTextArea element with values 1`] = `
<div class="">
  <label class="o-forms-field" data-biz-ops-type="textarea">
    <span class="o-forms-title">
      <span class="o-forms-title__main" id="inline-radio-round-group-title">LABEL:</span>
      <span class="o-forms-title__prompt description-text">
        <span>DESCRIPTION</span>
      </span>
    </span>
    <span class="o-forms-input o-forms-input--textarea">
      <textarea name="PROPERTY NAME" id="id-PROPERTY NAME" rows="8">VALUE</textarea>
    </span>
  </label>
</div>
`;
